psycop/common/model_training/data_loader/data_loader.py

```diff
@@ -42,16 +42,97 @@ def _check_column_names(self, df: pd.DataFrame):
         else:
             log.debug("No column name checker specified. Skipping column name check.")
 
+    @staticmethod
+    def _check_dataframes_can_be_concatenated(
+        datasets: list[pd.DataFrame],
+        uuid_column: str,
+    ) -> bool:
+        """Check if pred_time_uuid columns are sorted so they can be concatenated
+        instead of joined."""
+        base_uuid = datasets[0][uuid_column]
+        return all(base_uuid.equals(df[uuid_column]) for df in datasets[1:])
+
+    @staticmethod
+    def _check_dataframes_can_be_joined(
+        datasets: list[pd.DataFrame],
+        uuid_column: str,
+    ) -> bool:
+        """Check if pred_time_uuid columns contain the same elements so they can
+        be joined without introducing new rows"""
+        base_uuid = datasets[0][uuid_column]
+        return all(base_uuid.isin(df[uuid_column]).all() for df in datasets[1:])
+
+    @staticmethod
+    def _remove_id_columns(
+        datasets: list[pd.DataFrame],
+        id_columns: Sequence[str],
+    ) -> list[pd.DataFrame]:
+        """Remove id columns from all but the first dataset"""
+        return [
+            dataset.drop(columns=id_columns) if i > 0 else dataset
+            for i, dataset in enumerate(datasets)
+        ]
+
+    def _check_and_merge_feature_sets(
+        self,
+        datasets: list[pd.DataFrame],
+    ) -> pd.DataFrame:
+        """Check if datasets can be concatenated or need to be joined."""
+        n_rows = [dataset.shape[0] for dataset in datasets]
+        if len(set(n_rows)) != 1:
+            raise ValueError(
+                "The datasets have a different amount of rows. "
+                + "Ensure that they have been created with the same "
+                + "prediction times.",
+            )
+        shared_id_columns = [
+            self.data_cfg.col_name.id,
+            self.data_cfg.col_name.pred_time_uuid,
```

Comment:
This col is the concate of the id and pred_timestamp cols, so we only need to check that one :-)

Reply:
Only checking that one (see `_check_dataframes_can_be_concatenated`) - these are the columns that need to be dropped for concatenation :)